src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseFileReadinessServiceTest.kt

```diff
@@ -0,0 +1,123 @@
+    private lateinit var printResponseFileReadinessService: PrintResponseFileReadinessService
+
+    companion object {
+        const val DIRECTORY = "/some-directory"
+    }
+
+    @Test
+    fun `should not mark print response file and not send to queue when no files are present`() {
+        // Given
+
+        given(sftpProperties.printResponseDownloadDirectory).willReturn(DIRECTORY)
+        given(sftpService.identifyFilesToBeProcessed(any())).willReturn(emptyList())
+
+        // When
+        printResponseFileReadinessService.markPrintResponseFileForProcessing()
+
+        // Then
+        verify(sftpService).identifyFilesToBeProcessed(DIRECTORY)
+        verifyNoInteractions(printMessagingService)
+        verifyNoMoreInteractions(sftpService)
+    }
+
+    @Test
+    fun `should mark print response file and send to queue when files are present`() {
+        // Given
+        val matchingFileName1 = "status-20221101171156056.json"
+        val matchingFileName2 = "status-20221101171156057.json"
+        val renamedFileName1 = "$matchingFileName1.processing"
+        val renamedFileName2 = "$matchingFileName2.processing"
+
+        given(sftpProperties.printResponseDownloadDirectory).willReturn(DIRECTORY)
+        given(sftpService.identifyFilesToBeProcessed(any())).willReturn(listOf(matchingFileName1, matchingFileName2))
+        given(sftpService.markFileForProcessing(eq(DIRECTORY), eq(matchingFileName1))).willReturn(renamedFileName1)
+        given(sftpService.markFileForProcessing(eq(DIRECTORY), eq(matchingFileName2))).willReturn(renamedFileName2)
+
+        // When
+        printResponseFileReadinessService.markPrintResponseFileForProcessing()
+
+        // Then
+        verify(sftpService).identifyFilesToBeProcessed(DIRECTORY)
+        verify(sftpService).markFileForProcessing(DIRECTORY, matchingFileName1)
+        verify(sftpService).markFileForProcessing(DIRECTORY, matchingFileName2)
+        verify(printMessagingService).submitPrintResponseFileForProcessing(
+            ProcessPrintResponseFileMessage(DIRECTORY, renamedFileName1)
+        )
+        verify(printMessagingService).submitPrintResponseFileForProcessing(
+            ProcessPrintResponseFileMessage(DIRECTORY, renamedFileName2)
+        )
+        verifyNoMoreInteractions(sftpService, printMessagingService)
+    }
+
+    @Test
+    fun `should continue marking print response file and sending to queue when some of files marking fails due to exception`() {
+        // Given
+        val matchingFile1ThatThrowsException = "status-20221101171156051.json"
+        val matchingFile3ThatThrowsException = "status-20221101171156053.json"
+        val matchingFileName2 = "status-20221101171156052.json"
+        val matchingFileName4 = "status-20221101171156054.json"
+        val renamedFileName2 = "$matchingFileName2.processing"
+        val renamedFileName4 = "$matchingFileName4.processing"
+        val expectedExceptionThrown = RuntimeException("Some error")
```

Comment:
Re my comments above re: exception handling etc ... this is exactly why ....
Reading the tests should help document and help you understand the implementation.
Looking at this test I can see that we expect an exception to be thrown, but I don't know why or what it means. Under what circumstances would an exception be thrown in a running environment?

(playing devil's advocate here - clearly I know why the exception is thrown because its in the context of your PR description etc 😁  But will we remember this when reading the code in 3 months time? Will a new developer grok this from reading the test? ¯\_(ツ)_/¯ )